src/sentry/sentry_apps/token_exchange/grant_exchanger.py

```diff
@@ -108,18 +108,28 @@ def grant(self) -> ApiGrant:
                 .get(code=self.code)
             )
         except ApiGrant.DoesNotExist:
-            raise APIUnauthorized("Could not find grant")
+            raise SentryAppIntegratorError(
+                "Could not find grant for given code",
+                webhook_context={"code": self.code[:4]},
```

Comment:
is it useful to only include the first few characters of the code? what is the integrator expected to do with this?

Reply:
is there anything that the integrator stores on their side that we're also storing? that would be the more useful perhaps

Reply:
:0) good idea. Turns out in the oauth process we send the SentryAppInstallation uuid which is stored on the 3p side. Trying to determine how safe exposing the installation uuid is though